### PR TITLE
[feature](Cloud) Add one config to control FE's startup with or without built in vault

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 1]
```diff
@@ -1841,6 +1841,18 @@ void MetaServiceImpl::get_cluster(google::protobuf::RpcController* controller,
         return;
     }
 
+    while (true) {
```
Comment (w41ter) [inline cloud/src/meta-service/meta_service_resource.cpp:1844]: It seems the `while` here is not necessary.

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: cloud/src/meta-service/meta_service.cpp [hunk 2]
```diff
@@ -560,6 +560,11 @@ void MetaServiceImpl::create_tablets(::google::protobuf::RpcController* controll
             return;
         }
 
+        // This instance hasn't enable storage vault which means it's using legacy cloud mode
+        if (!instance.enable_storage_vault()) {
```
Comment (platoneko) [inline cloud/src/meta-service/meta_service.cpp:564]: 加个 DCHECK 吧，按理说 FE 在 `!enable_storage_vault` 时不应该 `has_storage_vault_name`

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 3]
```diff
@@ -400,9 +397,7 @@ static int add_hdfs_storage_vault(InstanceInfoPB& instance, Transaction* txn,
     storage_vault_key({instance.instance_id(), vault_id}, &key);
     hdfs_param.set_id(vault_id);
     if (vault_id == BUILT_IN_STORAGE_VAULT_ID) {
-        hdfs_param.set_name(BUILT_IN_STORAGE_VAULT_NAME);
-        instance.set_default_storage_vault_name(BUILT_IN_STORAGE_VAULT_NAME);
-        instance.set_default_storage_vault_id(BUILT_IN_STORAGE_VAULT_ID);
+        hdfs_param.set_name(BUILT_IN_STORAGE_VAULT_NAME.data());
```
Comment (platoneko) [inline cloud/src/meta-service/meta_service_resource.cpp:400]: BUILT_IN_STORAGE_VAULT_ID 可能用不到，考虑直接在 create instance 函数给 vault 设置 name = BUILT_IN_STORAGE_VAULT_NAME

**Review by platoneko — COMMENTED** [review]
FE 需要有个 Env 级属性 enableStorageVault，启动时获取，create tablets 需要用到来决定请求是否带上 vault name

**Comment by ByteYue** [comment]
> FE 需要有个 Env 级属性 enableStorageVault，启动时获取，create tablets 需要用到来决定请求是否带上 vault name

估计得在get cluster时获取local type里拿到这个属性

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by platoneko — APPROVED** [review]

**Review by w41ter — APPROVED** [review]

**Pull request merged**